### PR TITLE
Sanitize CHOMP initialization method parameter

### PR DESCRIPTION
**Pull request by onionsflying**
Return false when interpolation method is invalid.

Edit by v4hn: Additionally verify parameter during planner initialization.

### CODE REVIEW

**Comment by v4hn** [comment]
Thanks for your contribution @onionsflying .
I extended your one-line patch to validate parameters during setup already.

Please verify this works for you too.

**Comment by onionsflying** [comment]
Thanks, it looks good to me. @v4hn

File: moveit_planners/chomp/chomp_optimizer_adapter/src/chomp_optimizer_adapter.cpp [hunk 1]
```diff
@@ -150,12 +150,18 @@ class OptimizerAdapter : public planning_request_adapter::PlanningRequestAdapter
...
+    params_.trajectory_initialization_method_ = std::string("fillTrajectory");
+    std::string trajectory_initialization_method;
+    if (!nh.getParam("trajectory_initialization_method", trajectory_initialization_method))
     {
-      params_.trajectory_initialization_method_ = std::string("fillTrajectory");
-      ROS_INFO_STREAM("Param trajectory_initialization_method was not set. Using New value as: "
+      ROS_INFO_STREAM("Param trajectory_initialization_method was not set. Using value: "
                       << params_.trajectory_initialization_method_);
     }
+    else if (!params_.setTrajectoryInitializationMethod(trajectory_initialization_method))
```
Comment (simonschmeisser) [inline moveit_planners/chomp/chomp_optimizer_adapter/src/chomp_optimizer_adapter.cpp:161]: can we have brackets here as well?

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_parameters.cpp [hunk 2]
```diff
@@ -80,4 +80,18 @@ void ChompParameters::setRecoveryParams(double learning_rate, double ridge_facto
...
   this->max_iterations_ = max_iterations;
 }
+
+const std::vector<std::string> ChompParameters::VALID_INITIALIZATION_METHODS{ "quintic-spline", "linear", "cubic",
+                                                                              "fillTrajectory" };
+
+bool ChompParameters::setTrajectoryInitializationMethod(std::string method)
+{
+  if (std::find(VALID_INITIALIZATION_METHODS.cbegin(), VALID_INITIALIZATION_METHODS.cend(), method) !=
+      VALID_INITIALIZATION_METHODS.end())
```
Comment (simonschmeisser) [inline moveit_planners/chomp/chomp_motion_planner/src/chomp_parameters.cpp:90]:
```suggestion
  if (VALID_INITIALIZATION_METHODS.count(method) != 1)
```
could be used if you switch from `std::vector<std::string>` to `std::set<std::string>` (or `.contains(method)` starting with c++ 20 ...)

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_parameters.cpp [hunk 3]
```diff
@@ -80,4 +80,18 @@ void ChompParameters::setRecoveryParams(double learning_rate, double ridge_facto
...
+  {
+    this->trajectory_initialization_method_ = std::move(method);
```
Comment (simonschmeisser) [inline moveit_planners/chomp/chomp_motion_planner/src/chomp_parameters.cpp:92]:
```suggestion
    this->trajectory_initialization_method_ = method;
```
if you pass a `const std::string& method` you can micro-optimize even more :grin:

**Review by simonschmeisser — APPROVED** [review]
Thanks for your PR! Don't feel annoyed by my nitpicking please

**Comment by simonschmeisser** [comment]
Oh and one more thing: there's an edit link somewhere on this page and you could use it to set a more descriptive PR title (you could copy the commit message)

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_parameters.cpp (hunk 2 above)
Comment (v4hn) [reply moveit_planners/chomp/chomp_motion_planner/src/chomp_parameters.cpp:90]: We (sadly) don't have C++20, or we could improve so much code with Ranges... :smile: 

I don't see why I should use a rb-tree for a list of 4 elements, so I will keep this as is.

File: moveit_planners/chomp/chomp_optimizer_adapter/src/chomp_optimizer_adapter.cpp (hunk 1 above)
Comment (v4hn) [reply moveit_planners/chomp/chomp_optimizer_adapter/src/chomp_optimizer_adapter.cpp:161]: done

**Review by henningkayser — APPROVED** [review]

**Pull request merged**